src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
_UPDATE:_

Issuing a HEAD http request to the "attachment URL" will provide the header information that we desire.  We can get the `content-type`, `content-length`, and `content-md5` from this call.  Obviously, it is still an additional http request but it returns a minimal set of data as compared to the `.fetch()` GET request which returns the entire document.

Still left to do is to confirm that we can gleen the `revpos` correctly in every case but I think that should be possible.  We'll handle this piece in a future PR.